addons/web/static/src/views/fields/many2many_tags/many2many_color_dot_field.js

```diff
@@ -0,0 +1,74 @@
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { evaluateBooleanExpr } from "@web/core/py_js/py";
+import { Many2ManyTagsField, many2ManyTagsField } from "./many2many_tags_field";
+import { BadgeTagDot } from "@web/core/tags_list/badge_tag_dot";
+
+/**
+ * Extension of the `many2many_tags` widget in which the color field is a
+ * VARCHAR accepting a hexadecimal color code.
+ *
+ * Instead of coloring the entire tag background (default behavior), this widget
+ * displays a colored bullet (dot) placed before the tag label. This allows the
+ * tag to keep its regular appearance while visually indicating its color
+ * through the small dot.
+ */
+class Many2ManyTagsColorDotField extends Many2ManyTagsField {
+    static components = {
+        ...Many2ManyTagsField.components,
+        Tag: BadgeTagDot,
+    };
+
+    /** @override */
+    getTagProps(record) {
+        const value = record.data[this.props.colorField];
+        const hexRegex = /^#([0-9A-Fa-f]{3}|[0-9A-Fa-f]{6})$/;
+        const dotColor = hexRegex.test(value) ? value : "#000";
+        return {
+            ...super.getTagProps(record),
+            color: "basic", //Force grey background
+            dotColor,
+        };
+    }
+}
+
+const supportedOptions = many2ManyTagsField.supportedOptions
+    .filter((option) => option.name !== "on_tag_click")
+    .map((option) => {
+        if (option.name === "color_field") {
+            return {
+                ...option,
+                label: _t("Color dot field"),
+                help: _t("Set a char field that colors the dot on the tag."),
+                availableTypes: ["char"],
+            };
+        }
+        return { ...option };
+    });
+supportedOptions.push({
+    name: "can_edit",
+    label: _t("Open form on tag click"),
+    type: "boolean",
+});
+
+export const many2ManyTagsColorDotField = {
+    ...many2ManyTagsField,
+    component: Many2ManyTagsColorDotField,
+    displayName: _t("Color dot tags"),
+    supportedOptions,
+    additionalClasses: ["o_field_many2many_tags"],
+    extractProps({ attrs, options, string, placeholder }, dynamicInfo) {
+        const hasEditPermission = attrs.can_write ? evaluateBooleanExpr(attrs.can_write) : true;
+        const props = many2ManyTagsField.extractProps(
+            { attrs, options, string, placeholder },
+            dynamicInfo
+        );
+        delete props.onTagClick;
+        if (hasEditPermission && options.can_edit) {
```

Comment:
I know I told you the name "can_edit", but it's bad. There's the `can_write` attribute and the `can_edit` option... Actually, opening the record isn't necessarily linked to the write access rights. You may open it in readonly if you don't have right access. So maybe
- rename into `can_open` or somehting like that (in many2one, the option is named `no_open`, but we don't like double negation so...)
- do not prevent opening if can_write false.

What do you think?